extensions/src/main/scala/ai/h2o/sparkling/extensions/internals/CollectCategoricalDomainsTask.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.extensions.internals;
+
+import java.util.Arrays;
+import jsr166y.ForkJoinTask;
+import water.H2O;
+import water.Key;
+import water.MRTask;
+import water.parser.BufferedString;
+import water.parser.PackedDomains;
+import water.util.Log;
+
+public class CollectCategoricalDomainsTask extends MRTask<CollectCategoricalDomainsTask> {
+  private final Key frameKey;
+  private byte[][] packedDomains;
+
+  public CollectCategoricalDomainsTask(Key frameKey) {
+    this.frameKey = frameKey;
+  }
+
+  @Override
+  public void setupLocal() {
+    if (!LocalNodeDomains.containsDomains(frameKey)) return;
+    final String[][][] localDomains = LocalNodeDomains.getDomains(frameKey);
+    if (localDomains.length == 0) return;
+    packedDomains = chunkDomainsToPackedDomains(localDomains[0]);
+    for (int i = 1; i < localDomains.length; i++) {
+      byte[][] anotherPackedDomains = chunkDomainsToPackedDomains(localDomains[i]);
+      packedDomains = mergePackedDomains(packedDomains, anotherPackedDomains);
```

Comment:
Isn't the assignment extra `packedDomains = ?` Idea also gives warning about this

Reply:
yes, you're right, since the first argument is altered and returned. will update thanks!